core/primitives/src/views.rs

```diff
@@ -1745,11 +1745,11 @@ pub enum TxExecutionStatus {
     Final,
 }
 
-#[derive(BorshSerialize, BorshDeserialize, serde::Serialize, serde::Deserialize, Debug, Clone)]
+#[derive(serde::Serialize, serde::Deserialize, Debug, Clone)]
 #[serde(untagged)]
 pub enum FinalExecutionOutcomeViewEnum {
-    FinalExecutionOutcome(FinalExecutionOutcomeView),
     FinalExecutionOutcomeWithReceipt(FinalExecutionOutcomeWithReceiptView),
+    FinalExecutionOutcome(FinalExecutionOutcomeView),
```

Comment:
Could you please a comment here mentioning that the ordering is relevant and why.

Reply:
I did, plus added tests that I think are useful